src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1327,11 +1327,23 @@ private void PrepareContainers()
                 throw new InvalidOperationException();
             }
 
-            var nameSuffix = GetRandomNameSuffix();
+            var nameSuffix = string.Empty;
+
+            if (container.GetContainerLifetimeType() == ContainerLifetimeType.Default)
```

Comment:
@ReubenBond we need to make sure this doesn't regress testing scenarios right?

Reply:
I've been hesitant to add tests to this PR for persistent resources as I don't want to introduce any test instability by leaving resources behind after a test run.